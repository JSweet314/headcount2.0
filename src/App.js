import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CardsContainer from './CardsContainer';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      districts: props.districts,
      searchValue: ''
    };
  }

  searchDistrict = (e) => {
    this.setState({
      searchValue: e.target.value.toUpperCase()
    });
  } 

  render() {
    return (
      <div>
        <input 
          type='search' 
          value={this.state.searchValue} 
          onChange={this.searchDistrict} />
        <CardsContainer districts={ this.state.districts } searchValue={this.state.searchValue} />
      </div>
    );
  }
}

App.propTypes = {
  districts: PropTypes.objectOf(PropTypes.object).isRequired
};

export default App;
