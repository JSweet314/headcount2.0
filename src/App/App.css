.App {
	height: 100%;
}

.Cards-container {
	margin-top: 150px;
}