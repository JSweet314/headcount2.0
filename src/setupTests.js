import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter() });

export const kinderData = [
  {
    "Location": "Colorado",
    "TimeFrame": 2007,
    "DataFormat": "Percent",
    "Data": 0.39465
  },
  {
    "Location": "Colorado",
    "TimeFrame": 2006,
    "DataFormat": "Percent",
    "Data": 0.33677
  },
  {
    "Location": "Colorado",
    "TimeFrame": 2005,
    "DataFormat": "Percent",
    "Data": 0.27807
  }
];