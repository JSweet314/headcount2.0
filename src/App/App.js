import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CardsContainer from '../CardsContainer/CardsContainer.js';
import Search from '../Search/Search.js';
import './App.css';
import '../index.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      districts: props.districts,
      searchValue: '',
      selectedLocations: []
    };
  }

  searchDistrict = (event) => {
    this.setState({
      searchValue: event.target.value
    });
  } 

  clearSearch = () => {
    this.setState({searchValue: ''});
  }

  selectLocation = (location) => {
    const searchLocation = this.state.districts.findByName(location);
    if (this.state.selectedLocations.includes(searchLocation)) {
      this.setState({ 
        selectedLocations: this.state.selectedLocations
          .filter(compareLocation => compareLocation !== searchLocation)});
      return;
    }
    if (this.state.selectedLocations.length >= 1) {
      this.setState({
        selectedLocations: [this.state.selectedLocations[0], searchLocation] 
      });
      return;
    }
    this.setState({ 
      selectedLocations: [searchLocation]
    });
  }

  render() {
    return (
      <div className="App">
        <Search 
          clearSearch={this.clearSearch}
          searchValue={this.state.searchValue}
          searchDistrict={this.searchDistrict} />
        <div className="Cards-container">
          <CardsContainer 
            districts={this.state.districts} 
            searchValue={this.state.searchValue}
            selectLocation={this.selectLocation} 
            selectedLocations={this.state.selectedLocations} />
        </div>
      </div>
    );
  }
}

App.propTypes = {
  districts: PropTypes.objectOf(PropTypes.object).isRequired
};

export default App;
