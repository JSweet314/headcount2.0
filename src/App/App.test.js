import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import DistrictRepository from '../helper.js';
import {kinderData} from '../setupTests.js';
import {shallow} from 'enzyme';

describe('App', () => {
  let app;
  const districts = new DistrictRepository(kinderData);

  beforeEach(() => {
    app = shallow(<App districts={districts}/>);
  });


  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<App districts={districts} />, div);
  });

  it('should render the app', () => {
    expect(app).toMatchSnapshot();
  });

  it('should hold an object of district data in state', () => {
    expect(app.state('districts')).toBeDefined();
  });
});


