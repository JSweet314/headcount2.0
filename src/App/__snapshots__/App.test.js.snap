// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    districts={
        DistrictRepository {
            "stats": Object {
              "ACADEMY 20": Object {
                "location": "ACADEMY 20",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
              "COLORADO": Object {
                "location": "COLORADO",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
              "COLORADO SPRINGS": Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Search
          clearSearch={[Function]}
          searchDistrict={[Function]}
          searchValue=""
/>,
        <CardsContainer
          districts={
                    DistrictRepository {
                              "stats": Object {
                                "ACADEMY 20": Object {
                                  "location": "ACADEMY 20",
                                  "stats": Object {
                                    "2005": 0.278,
                                    "2006": 0.337,
                                    "2007": 0.395,
                                  },
                                },
                                "COLORADO": Object {
                                  "location": "COLORADO",
                                  "stats": Object {
                                    "2005": 0.278,
                                    "2006": 0.337,
                                    "2007": 0.395,
                                  },
                                },
                                "COLORADO SPRINGS": Object {
                                  "location": "COLORADO SPRINGS",
                                  "stats": Object {
                                    "2005": 0.278,
                                    "2006": 0.337,
                                    "2007": 0.395,
                                  },
                                },
                              },
                            }
          }
          searchValue=""
          selectLocation={[Function]}
          selectedLocations={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clearSearch": [Function],
          "searchDistrict": [Function],
          "searchValue": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "districts": DistrictRepository {
            "stats": Object {
              "ACADEMY 20": Object {
                "location": "ACADEMY 20",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
              "COLORADO": Object {
                "location": "COLORADO",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
              "COLORADO SPRINGS": Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2005": 0.278,
                  "2006": 0.337,
                  "2007": 0.395,
                },
              },
            },
          },
          "searchValue": "",
          "selectLocation": [Function],
          "selectedLocations": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Search
            clearSearch={[Function]}
            searchDistrict={[Function]}
            searchValue=""
/>,
          <CardsContainer
            districts={
                        DistrictRepository {
                                    "stats": Object {
                                      "ACADEMY 20": Object {
                                        "location": "ACADEMY 20",
                                        "stats": Object {
                                          "2005": 0.278,
                                          "2006": 0.337,
                                          "2007": 0.395,
                                        },
                                      },
                                      "COLORADO": Object {
                                        "location": "COLORADO",
                                        "stats": Object {
                                          "2005": 0.278,
                                          "2006": 0.337,
                                          "2007": 0.395,
                                        },
                                      },
                                      "COLORADO SPRINGS": Object {
                                        "location": "COLORADO SPRINGS",
                                        "stats": Object {
                                          "2005": 0.278,
                                          "2006": 0.337,
                                          "2007": 0.395,
                                        },
                                      },
                                    },
                                  }
            }
            searchValue=""
            selectLocation={[Function]}
            selectedLocations={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clearSearch": [Function],
            "searchDistrict": [Function],
            "searchValue": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "districts": DistrictRepository {
              "stats": Object {
                "ACADEMY 20": Object {
                  "location": "ACADEMY 20",
                  "stats": Object {
                    "2005": 0.278,
                    "2006": 0.337,
                    "2007": 0.395,
                  },
                },
                "COLORADO": Object {
                  "location": "COLORADO",
                  "stats": Object {
                    "2005": 0.278,
                    "2006": 0.337,
                    "2007": 0.395,
                  },
                },
                "COLORADO SPRINGS": Object {
                  "location": "COLORADO SPRINGS",
                  "stats": Object {
                    "2005": 0.278,
                    "2006": 0.337,
                    "2007": 0.395,
                  },
                },
              },
            },
            "searchValue": "",
            "selectLocation": [Function],
            "selectedLocations": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
